### PR TITLE
[IOTDB-611] Add documents introducing Data Query design fundamentals

### PR DESCRIPTION
**Pull request by wshao08**
JIRA issue: https://issues.apache.org/jira/projects/IOTDB/issues/IOTDB-611?filter=allopenissues

This PR is to add a design fundamental document in order to provide an entry level guide for designers and developers.

### CODE REVIEW

File: site/src/main/.vuepress/config.js [hunk 1]
```diff
@@ -438,8 +438,8 @@ var config = {
 						title: '5-DataQuery',
 						children: [
 							['5-DataQuery/1-DataQuery','DataQuery'],
-							['5-DataQuery/2-SeriesReader','SeriesReader'],
-							['5-DataQuery/3-ModificationHandle','ModificationHandle'],
+							['5-DataQuery/2-QueryFundamentals','QueryFundamentals']
+							['5-DataQuery/3-SeriesReader','SeriesReader'],
 							['5-DataQuery/4-RawDataQuery','RawDataQuery'],
```
Comment (HTHou) [inline site/src/main/.vuepress/config.js:443]: I think you miss a comma after`'QueryFundamentals']`.

File: site/src/main/.vuepress/config.js [hunk 2]
```diff
@@ -853,8 +853,8 @@ var config = {
 						title: '5-数据查询',
 						children: [
 							['5-DataQuery/1-DataQuery','数据查询'],
-							['5-DataQuery/2-SeriesReader','查询基础组件'],
-							['5-DataQuery/3-ModificationHandle','查询中的数据修改处理'],
+							['5-DataQuery/2-QueryFundamentals','查询基础介绍']
+							['5-DataQuery/3-SeriesReader','查询基础组件'],
```
Comment (HTHou) [inline site/src/main/.vuepress/config.js:857]: Missing comma after `'查询基础介绍']`.

File: site/src/main/.vuepress/config.js (hunk 1 above)
Comment (wshao08) [reply site/src/main/.vuepress/config.js]: Fixed, thanks

File: site/src/main/.vuepress/config.js (hunk 2 above)
Comment (wshao08) [reply site/src/main/.vuepress/config.js]: Fixed

File: docs/zh/SystemDesign/5-DataQuery/1-DataQuery.md [hunk 3]
```diff
@@ -35,8 +35,8 @@
 
 ## 相关文档
 
-* [基础查询组件](/zh/SystemDesign/5-DataQuery/2-SeriesReader.html)
-* [查询中的数据删改处理](/zh/SystemDesign/5-DataQuery/3-ModificationHandle.html)
+* [查询基础介绍](/SystemDesign/5-DataQuery/2-QueryFundamentals.html)
```
Comment (SailVR) [inline docs/zh/SystemDesign/5-DataQuery/1-DataQuery.md:38]:
```suggestion
* [查询基础介绍](/zh/SystemDesign/5-DataQuery/2-QueryFundamentals.html)
```

File: docs/zh/SystemDesign/5-DataQuery/2-QueryFundamentals.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+对于顺序和乱序文件的数据，其数据在文件中的分部特征有所不同。
+顺序文件的TimeseriesMetadata中所包含的ChunkMetadata也是有序的，也就是说如果按照chunkMetadata1, chunkMetadata2的顺序存储，那么将会保证chunkMetadata1.endtime <= chunkMetadata2.startTime。
+
+乱序文件的TimeseriesMetadata中所包含的ChunkMetadata是无序的，乱序文件中多个Chunk所覆盖的数据可能存在重叠，同时也可能与顺序文件中的chunk数据存在重叠。
+
+每个Chunk结构内部所包含的Page数据总是有序的，不管是从属于顺序文件还是乱序文件。也就是说前一个Page的最大时间戳不小于后一个的最小时间戳。因此在查询过程中可以充分利用这种有序性，通过统计信息对Page数据进行提前筛选。
+
+
+
+## 查询中的数据修改处理
```
Comment (qiaojialin) [inline docs/zh/SystemDesign/5-DataQuery/2-QueryFundamentals.md:61]: 现在3-ModificationHandle 还没删吧，跟 3-SeriesReader 编号冲突了，或者把 ModificationHandle 里的移动到这里来，或者把ModificationHandle和之后的往后顺移一位。

**Pull request merged**